### PR TITLE
Improve `array_prepend` signature for null and empty array

### PR DESCRIPTION
**Pull request by jayzhan211**
## Which issue does this PR close?

Closes #.
Part of #7142

### CODE REVIEW

File: datafusion/sqllogictest/test_files/array.slt [hunk 1]
```diff
@@ -1440,18 +1440,58 @@ select array_append(column1, make_array(1, 11, 111)), array_append(make_array(ma
...
+[, 1]
+
+query ?
+select array_prepend(null, [[1,2,3]]);
+----
+[, [1, 2, 3]]
+
+# DuckDB: [[]]
+# ClickHouse: [[]]
+# TODO: We may also return [[]]
```
Comment (jayzhan211) [inline datafusion/sqllogictest/test_files/array.slt:1480]: Unsure should we implement this, return value with larger dimension seems weird to me.

File: datafusion/sqllogictest/test_files/array.slt [hunk 2]
```diff
@@ -1440,18 +1440,58 @@ select array_append(column1, make_array(1, 11, 111)), array_append(make_array(ma
...
+query error
+select array_prepend([], []);
+
+# DuckDB: [null]
+# ClickHouse: [null]
+# TODO: We may also return [null]
```
Comment (jayzhan211) [inline datafusion/sqllogictest/test_files/array.slt:1486]: Same with this one.

Comment (jayzhan211) [reply datafusion/sqllogictest/test_files/array.slt:1486]: Both cases are not considered in `array_append`

**Review by Weijun-H — APPROVED** [review]
LGTM! Thanks @jayzhan211 👍 
And I could try to test the null for `LargeList` after #8569 merged.

File: datafusion/sqllogictest/test_files/array.slt (hunk 1 above)
Comment (alamb) [reply datafusion/sqllogictest/test_files/array.slt:1480]: I agree -- let's not implement this until someone asks for it explicitly

**Review by alamb — APPROVED** [review]
Thank you @jayzhan211  -- this is looking very nice. 👌

**Comment by alamb** [comment]
Thank you for the review @Weijun-H

**Pull request merged**